// @react-spectrum/accordion/dist/Accordion.module.js
import "./vars.77b903e1.css";
import $aMy5K$accordion_vars_cssmodulejs from "./accordion_vars_css.module.js";
import {UNSTABLE_DisclosureGroup as $aMy5K$UNSTABLE_DisclosureGroup, UNSTABLE_Disclosure as $aMy5K$UNSTABLE_Disclosure, UNSTABLE_DisclosurePanel as $aMy5K$UNSTABLE_DisclosurePanel, Heading as $aMy5K$Heading, Button as $aMy5K$Button} from "react-aria-components";
import $aMy5K$spectrumiconsuiChevronLeftMedium from "@spectrum-icons/ui/ChevronLeftMedium";
import $aMy5K$spectrumiconsuiChevronRightMedium from "@spectrum-icons/ui/ChevronRightMedium";
import {useStyleProps as $aMy5K$useStyleProps, useDOMRef as $aMy5K$useDOMRef, classNames as $aMy5K$classNames} from "@react-spectrum/utils";
import $aMy5K$react, {forwardRef as $aMy5K$forwardRef} from "react";
import {useLocale as $aMy5K$useLocale} from "@react-aria/i18n";
import {useProviderProps as $aMy5K$useProviderProps} from "@react-spectrum/provider";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 







function $2a5a5b7faca81754$var$Accordion(props, ref) {
    props = (0, $aMy5K$useProviderProps)(props);
    let { styleProps: styleProps } = (0, $aMy5K$useStyleProps)(props);
    let domRef = (0, $aMy5K$useDOMRef)(ref);
    return /*#__PURE__*/ (0, $aMy5K$react).createElement((0, $aMy5K$UNSTABLE_DisclosureGroup), {
        ...props,
        ...styleProps,
        ref: domRef,
        className: (0, $aMy5K$classNames)((0, ($parcel$interopDefault($aMy5K$accordion_vars_cssmodulejs))), 'spectrum-Accordion', styleProps.className)
    }, props.children);
}
function $2a5a5b7faca81754$var$Disclosure(props, ref) {
    props = (0, $aMy5K$useProviderProps)(props);
    let domRef = (0, $aMy5K$useDOMRef)(ref);
    return /*#__PURE__*/ (0, $aMy5K$react).createElement((0, $aMy5K$UNSTABLE_Disclosure), {
        ...props,
        ref: domRef,
        className: ({ isExpanded: isExpanded, isDisabled: isDisabled })=>(0, $aMy5K$classNames)((0, ($parcel$interopDefault($aMy5K$accordion_vars_cssmodulejs))), 'spectrum-Accordion-item', {
                'is-expanded': isExpanded,
                'is-disabled': isDisabled
            })
    }, props.children);
}
function $2a5a5b7faca81754$var$DisclosurePanel(props, ref) {
    let domRef = (0, $aMy5K$useDOMRef)(ref);
    return /*#__PURE__*/ (0, $aMy5K$react).createElement((0, $aMy5K$UNSTABLE_DisclosurePanel), {
        ref: domRef,
        className: (0, $aMy5K$classNames)((0, ($parcel$interopDefault($aMy5K$accordion_vars_cssmodulejs))), 'spectrum-Accordion-itemContent'),
        ...props
    }, props.children);
}
function $2a5a5b7faca81754$var$DisclosureHeader(props, ref) {
    let { level: level = 3 } = props;
    let { direction: direction } = (0, $aMy5K$useLocale)();
    let domRef = (0, $aMy5K$useDOMRef)(ref);
    return /*#__PURE__*/ (0, $aMy5K$react).createElement((0, $aMy5K$Heading), {
        ref: domRef,
        level: level,
        className: (0, $aMy5K$classNames)((0, ($parcel$interopDefault($aMy5K$accordion_vars_cssmodulejs))), 'spectrum-Accordion-itemHeading')
    }, /*#__PURE__*/ (0, $aMy5K$react).createElement((0, $aMy5K$Button), {
        slot: "trigger",
        className: ({ isHovered: isHovered, isFocusVisible: isFocusVisible })=>(0, $aMy5K$classNames)((0, ($parcel$interopDefault($aMy5K$accordion_vars_cssmodulejs))), 'spectrum-Accordion-itemHeader', {
                'is-hovered': isHovered,
                'focus-ring': isFocusVisible
            })
    }, direction === 'ltr' ? /*#__PURE__*/ (0, $aMy5K$react).createElement((0, $aMy5K$spectrumiconsuiChevronRightMedium), {
        "aria-hidden": "true",
        UNSAFE_className: (0, $aMy5K$classNames)((0, ($parcel$interopDefault($aMy5K$accordion_vars_cssmodulejs))), 'spectrum-Accordion-itemIndicator')
    }) : /*#__PURE__*/ (0, $aMy5K$react).createElement((0, $aMy5K$spectrumiconsuiChevronLeftMedium), {
        "aria-hidden": "true",
        UNSAFE_className: (0, $aMy5K$classNames)((0, ($parcel$interopDefault($aMy5K$accordion_vars_cssmodulejs))), 'spectrum-Accordion-itemIndicator')
    }), props.children));
}
/** A group of disclosures that can be expanded and collapsed. */ const $2a5a5b7faca81754$export$a766cd26d0d69044 = /*#__PURE__*/ (0, $aMy5K$forwardRef)($2a5a5b7faca81754$var$Accordion);
/** A collapsible section of content composed of a heading that expands and collapses a panel. */ const $2a5a5b7faca81754$export$74a362b31437ec83 = /*#__PURE__*/ (0, $aMy5K$forwardRef)($2a5a5b7faca81754$var$Disclosure);
/** The panel that contains the content of an disclosure. */ const $2a5a5b7faca81754$export$feabaa331e1d464c = /*#__PURE__*/ (0, $aMy5K$forwardRef)($2a5a5b7faca81754$var$DisclosurePanel);
/** The heading of the disclosure. */ const $2a5a5b7faca81754$export$1e037ddd32f5b8ac = /*#__PURE__*/ (0, $aMy5K$forwardRef)($2a5a5b7faca81754$var$DisclosureHeader);


export {$2a5a5b7faca81754$export$a766cd26d0d69044 as Accordion, $2a5a5b7faca81754$export$74a362b31437ec83 as Disclosure, $2a5a5b7faca81754$export$feabaa331e1d464c as DisclosurePanel, $2a5a5b7faca81754$export$1e037ddd32f5b8ac as DisclosureHeader};
